glean-core/src/metrics/experiment.rs

```diff
@@ -18,9 +18,25 @@ use crate::Lifetime;
 // An internal ping name, not to be touched by anything else
 const INTERNAL_STORAGE: &str = "glean_internal_info";
 
+// The following values are maximums related to experiments, these were
+// chosen based on current desktop usage. See here, for reference:
+// https://searchfox.org/mozilla-central/rev/3366c3d24f1c3818df37ec0818833bf085e41a53/toolkit/components/telemetry/app/TelemetryEnvironment.jsm#58
+
 /// The maximum length of the experiment id and the branch id. Identifiers
 /// longer than this number of characters are truncated.
-const MAX_EXPERIMENTS_IDS_LEN: usize = 30;
+const MAX_EXPERIMENTS_IDS_LEN: usize = 100;
```

Comment:
Let's not bump this limit and keep using this for experiment branch too. If that becomes a problem for the experiments library, we can discuss and bump.